Add ``array_position`` scalar function (#12249)
crate/crate: docs/general/builtins/scalar-functions.rst

    SELECT 1 row in set (... sec)

.. _scalar-array_position:

``array_position(anycompatiblearray, anycompatible [, integer ] ) → integer``
-----------------------------------------------------------------------------

The ``array_position`` function returns the position of the first
occurrence of the second argument in the ``array``, or ``NULL`` if it's not

> present. If the third argument is given, the search begins at that position.
> The third argument is ignored if it's null or not in within the ``array``
> range. It is also possible to search for ``NULL`` values.

Reviewer: ```suggestion
present. If the third argument is given, the search begins at that position.
The third argument is ignored if it's null. If not within the ``array`` range,
``NULL`` is returned. It is also possible to search for ``NULL`` values.
```

Reviewer: should be something like this, according to https://github.com/crate/crate/pull/11989#discussion_r782148204

Author: 👍

> The ``array_position`` function returns the subscript of the first
> occurrence of the second argument in the ``array``, or ``NULL`` if it's not
> present. If the third argument is given, the search begins at that subscript.
> The third argument is ignored if it's null or invalid value out of ``array``
> range. It is also possible to search for ``NULL`` values.

Reviewer 2: ```suggestion
The ``array_position`` function returns the position of the first
occurrence of the second argument in the ``array``, or ``NULL`` if it's not
present. If the third argument is given, the search begins at that position.
The third argument is ignored if it's null or not in within the ``array``
range. It is also possible to search for ``NULL`` values.
```

I don't think the `subscript` term fits here

::

    cr> SELECT array_position([1,3,7,4], 7) as position;
    +----------+
    | position |
    +----------+
    |        3 |
    +----------+
    SELECT 1 row in set (... sec)

> Begin the search from given subscript (optional).

Reviewer 2: ```suggestion
Begin the search from given position (optional).
```
